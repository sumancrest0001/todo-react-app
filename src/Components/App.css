body {
  font-size: 16px;
  margin: 0;
  padding: 0;
}

.app {
  width: 80%;
  padding: 20px;
  margin: auto;
}

.todo {
  padding: 10px;
  border: 1px solid rgb(43, 42, 42);
  border-radius: 20px;
  margin: 10px auto;
}

.input-field {
  width: 100%;
  display: flex;
  flex-direction: column;
  padding: 5px 0px;
  margin-bottom: 10px;
}

.input {
  padding: 5px;
  font-size: 16px;
}

.label {
  font-weight: 600;
  margin-bottom: 5px;
}

.is-invalid {
  border-color: red;
}

.radio-buttons {
  margin-bottom: 15px;
}

/* shared css */
.error {
  font-size: 14px;
  font-style: italic;
  color: red;
  padding: 5px;
}
.button {
  padding: 5px 10px;
}
