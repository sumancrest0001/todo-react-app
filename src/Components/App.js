import React, { useState } from 'react';
import FormScreen from './FormScreen';
import TodoList from './TodoList';
import ErrorBoundary from '../Components/Error/ErrorBoundary';
import './App.css';

function App() {
  const [screenOne, setScreenOne] = useState(false);
  const onScreenChange = () => {
    setScreenOne(!screenOne);
  };

  return (
    <div className="app">
      {
        screenOne ?
        (<ErrorBoundary>
          <FormScreen changeScreen={onScreenChange} />
        </ErrorBoundary>) :
        (<ErrorBoundary>
          <TodoList changeScreen={onScreenChange} />
        </ErrorBoundary>)
      }
    </div>
  );
}

export default App;
