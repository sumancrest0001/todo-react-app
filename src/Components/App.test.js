import React from 'react';
import renderer from 'react-test-renderer';

import App from './App';

it('renders correctly with out crash', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});