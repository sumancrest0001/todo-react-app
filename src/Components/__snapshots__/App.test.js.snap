// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with out crash 1`] = `
<div
  className="app"
>
  <div
    className="todolist"
  >
    <button
      className="todolist__button"
      onClick={[Function]}
      type="button"
    >
      Create new todo
    </button>
    <div
      className="todolist__filter-section"
    >
      <label>
        All
        <input
          checked={true}
          name="filter"
          onChange={[Function]}
          type="radio"
          value="all"
        />
      </label>
      <label>
        Completed
        <input
          checked={false}
          name="filter"
          onChange={[Function]}
          type="radio"
          value="completed"
        />
      </label>
      <label>
        Uncompleted
        <input
          checked={false}
          name="filter"
          onChange={[Function]}
          type="radio"
          value="incomplete"
        />
      </label>
    </div>
    <div
      className="todolist__todos"
    />
  </div>
</div>
`;
